code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForCredentials.java

```diff
@@ -0,0 +1,133 @@
+import java.util.Collections;
+import java.util.List;
+import com.google.api.client.json.webtoken.JsonWebToken;
+import com.google.api.client.json.webtoken.JsonWebSignature;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import java.util.Optional;
+
+public class CheckForCredentials {
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private static List<String> SCOPES;
+  private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+
+  public static Optional<Payload> getPayload(HttpServletRequest request, HttpServletResponse response) {
+
+    try {
+      NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+ 
+      InputStream in = CheckForCredentials.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+              if (in == null) {
+                  throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+              }
+
+      GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+      GoogleAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES).build();
+      ObjectMapper objectMapper = new ObjectMapper();
+      GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(HTTP_TRANSPORT, JSON_FACTORY)
+          .setAudience(Collections.singletonList("1080865471187-u1vse3ccv9te949244t9rngma01r226m"))
+          .build();
+
+        String idTokenString = request.getParameter("idtoken");
+        GoogleIdToken idToken = verifier.verify(idTokenString);
+
+        if (idToken != null) {
+            Payload payload = idToken.getPayload();
+            return Optional.of(payload);
+        }
+        return Optional.empty();
+    } catch (Exception e) {
+        e.printStackTrace();
+        return Optional.empty();
+    } 
+  }
+  //checks for credentials, if they don't exist, go create them
+  public static void createCredentials(HttpServletRequest request, HttpServletResponse response, List<String> scopes, String originalUrl, Optional<Payload> payload) throws IOException {
+    HashMap<String, String> userIdAndUrl = new HashMap<> ();
+    
+    try {
+      NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+ 
+      SCOPES = scopes;
+      InputStream in = CheckForCredentials.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
```

Comment:
This block of scopes appears twice, which is a good indication that it should be abstracted to a helper method.